tests/components/tts/test_init.py

```diff
@@ -716,3 +716,24 @@ async def test_setup_component_and_web_get_url_bad_config(hass, hass_client):
 
     req = await client.post(url, json=data)
     assert req.status == 400
+
+
+async def test_tags_with_wave(hass, demo_provider):
+    """Set up the demo platform and call service and receive voice."""
+
+    # below data represents an empty wav file
+    demo_data = bytes.fromhex(
+        "52 49 46 46 24 00 00 00 57 41 56 45 66 6d 74 20 10 00 00 00 01 00 02 00"
+        + "22 56 00 00 88 58 01 00 04 00 10 00 64 61 74 61 00 00 00 00"
+    )
+
+    tagged_data = tts.SpeechManager.write_tags(
+        "42f18378fd4393d18c8dd11d03fa9563c1e54491_en_-_demo.wav",
+        demo_data,
+        demo_provider,
+        "AI person is in front of your door.",
+        "en",
+        None,
+    )
+
+    assert tagged_data != demo_data
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)